tests/tests/mempool.rs

```diff
@@ -0,0 +1,396 @@
+        .pop()
+        .expect("Expected at least one transaction");
+
+    // Add transaction to mempool
+    eth_provider.mempool().unwrap().add_transaction(TransactionOrigin::Local, transaction.clone()).await.unwrap();
+
+    // Get the transaction event listener
+    let listener = eth_provider.mempool().unwrap().transaction_event_listener(transaction_signed.hash());
+
+    // Ensure the listener exists
+    assert!(listener.is_some());
+
+    // Check if the listener's hash matches the transaction's hash
+    assert_eq!(listener.unwrap().hash(), transaction_signed.hash());
+}
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_mempool_get_private_transactions(#[future] katana: Katana, _setup: ()) {
+    let eth_provider = katana.eth_provider();
+
+    // Create a sample transaction
+    let (transaction, transaction_signed) = create_sample_transactions(&katana, 1)
+        .await
+        .expect("Failed to create sample transaction")
+        .pop()
+        .expect("Expected at least one transaction");
+
+    // Add private transaction to mempool
+    eth_provider.mempool().unwrap().add_transaction(TransactionOrigin::Private, transaction.clone()).await.unwrap();
+
+    // Get private transactions
+    let private_transaction = eth_provider.mempool().unwrap().get_private_transactions();
+
+    // Check if the returned transaction hash matches
+    assert_eq!(*private_transaction[0].hash(), transaction_signed.hash());
+}
+
+// Helper function to create a sample transaction
+async fn create_sample_transactions(
+    katana: &Katana,
+    num_transactions: usize,
+) -> Result<Vec<(EthPooledTransaction, TransactionSigned)>, SignatureError> {
+    // Initialize a vector to hold transactions
+    let mut transactions = Vec::new();
+
+    for counter in 0..num_transactions {
+        // Get the Ethereum provider
+        let eth_provider = katana.eth_provider();
+
+        // Get the chain ID
+        let chain_id = eth_provider.chain_id().await.unwrap_or_default().unwrap_or_default().to();
+
+        // Create a new EIP-1559 transaction
+        let transaction = Transaction::Eip1559(TxEip1559 {
+            chain_id,
+            nonce: counter as u64,
+            gas_limit: 21000,
+            to: TxKind::Call(Address::random()),
+            value: U256::from(1000),
+            input: Bytes::default(),
+            max_fee_per_gas: 875_000_000,
+            max_priority_fee_per_gas: 0,
+            access_list: Default::default(),
+        });
+
+        // Sign the transaction
+        let signature = sign_message(katana.eoa().private_key(), transaction.signature_hash()).unwrap();
+
+        // Create a signed transaction
+        let transaction_signed = TransactionSigned::from_transaction_and_signature(transaction, signature);
+        // Recover the signer from the transaction
+        let signer = transaction_signed.recover_signer().ok_or(SignatureError::Recovery)?;
```

Comment:
I think you can do `katana.eoa().address()` instead of recovering the signer. Also, since this will be the same each loop, you can just do it outside the for loop